src/secp256k1.c

```diff
@@ -515,8 +514,25 @@ int secp256k1_ecdsa_sign(const secp256k1_context* ctx, secp256k1_ecdsa_signature
     secp256k1_scalar_clear(&msg);
     secp256k1_scalar_clear(&non);
     secp256k1_scalar_clear(&sec);
-    secp256k1_scalar_cmov(&r, &secp256k1_scalar_zero, !ret);
-    secp256k1_scalar_cmov(&s, &secp256k1_scalar_zero, !ret);
+    secp256k1_scalar_cmov(r, &secp256k1_scalar_zero, !ret);
+    secp256k1_scalar_cmov(s, &secp256k1_scalar_zero, !ret);
+    if (recid) {
+        const int zero = 0;
+        secp256k1_int_cmov(recid, &zero, !ret);
```

Comment:
Why? recid is not secret.

Reply:
The problem isn't `recid` being secret, it's `ret` being secret.

Reply:
I don't think there is much of an issue with declassifying ret. If the API is used correctly, ret is always 1 anyway.

Reply:
If we assume a user won't call `sign_*` without verifying the secret key first then yeah we can declassify `is_sec_valid` (which is what contaminates `ret`)
https://github.com/bitcoin-core/secp256k1/blob/a39c2b09de304b8f24716b59219ae37c2538c242/src/secp256k1.c#L510-L513

Reply:
@sipa This is not the scope of this PR. If we want to change this, I think that's a different discussion. 

The current signing code on master is overly cautious and assumes `is_sec_valid` to be a secret. That's exactly the reason why we cmov `r` and `s` (introduced by #710), see also the corresponding [code comment](https://github.com/bitcoin-core/secp256k1/pull/755/files/2876af4f8da952e39c06bc229d68cd4892ea2c85#diff-0d77700fa6e05639431c962ffaa5365aL510-L512) and the [exact reasoning in the PR](https://github.com/bitcoin-core/secp256k1/pull/710#discussion_r374510230).

Reply:
@real-or-random Thanks for the references, that makes sense. My point was mostly that at some point, the complexity of removing decreasingly relevant forms of non-ctness aren't worth it, but this isn't too bad yet.